app/components/markdown/markdown_latex_block/index.tsx

```diff
@@ -0,0 +1,232 @@
+// See LICENSE.txt for license information.
+
+import {useManagedConfig} from '@mattermost/react-native-emm';
+import Clipboard from '@react-native-community/clipboard';
+import React, {useCallback} from 'react';
+import {useIntl} from 'react-intl';
+import {Keyboard, View, Text, StyleSheet, Platform} from 'react-native';
+import MathView from 'react-native-math-view';
+import {useSafeAreaInsets} from 'react-native-safe-area-context';
+
+import FormattedText from '@components/formatted_text';
+import SlideUpPanelItem, {ITEM_HEIGHT} from '@components/slide_up_panel_item';
+import TouchableWithFeedback from '@components/touchable_with_feedback';
+import {Screens} from '@constants';
+import {bottomSheet, dismissBottomSheet, goToScreen} from '@screens/navigation';
+import {bottomSheetSnapPoint} from '@utils/helpers';
+import {getHighlightLanguageName} from '@utils/markdown';
+import {splitLatexCodeInLines} from '@utils/markdown/latex';
+import {preventDoubleTap} from '@utils/tap';
+import {changeOpacity, makeStyleSheetFromTheme} from '@utils/theme';
+
+const MAX_LINES = 2;
+
+type Props = {
+    content: string;
+    theme: Theme;
+}
+
+const getStyleSheet = makeStyleSheetFromTheme((theme: Theme) => {
+    const codeVerticalPadding = Platform.select({
+        ios: 4,
+        android: 0,
+    });
+
+    return {
+        bottomSheet: {
+            flex: 1,
+        },
+        container: {
+            borderColor: changeOpacity(theme.centerChannelColor, 0.15),
+            borderRadius: 3,
+            borderWidth: StyleSheet.hairlineWidth,
+            flexDirection: 'row',
+            flex: 1,
+        },
+        rightColumn: {
+            flexDirection: 'column',
+            flex: 1,
+            paddingLeft: 6,
+            paddingVertical: 4,
+        },
+        code: {
+            flexDirection: 'row',
+            justifyContent: 'flex-start',
+            marginLeft: 5,
+            paddingVertical: codeVerticalPadding,
+        },
+        plusMoreLinesText: {
+            color: changeOpacity(theme.centerChannelColor, 0.4),
+            fontSize: 11,
+            marginTop: 2,
+        },
+        language: {
+            alignItems: 'center',
+            backgroundColor: theme.sidebarHeaderBg,
+            justifyContent: 'center',
+            opacity: 0.8,
+            padding: 6,
+            position: 'absolute',
+            right: 0,
+            top: 0,
+        },
+        languageText: {
+            color: theme.sidebarHeaderTextColor,
+            fontSize: 12,
+        },
+        errorText: {
+            fontSize: 14,
+            marginHorizontal: 5,
+            color: theme.errorTextColor,
+        },
+    };
+});
+
+const LatexCodeBlock = ({content, theme}: Props) => {
+    const intl = useIntl();
+    const insets = useSafeAreaInsets();
+    const managedConfig = useManagedConfig<ManagedConfig>();
+    const styles = getStyleSheet(theme);
+
+    const handlePress = preventDoubleTap(() => {
```

Comment:
New function every render (may not be a problem if we trust enough the memoization done by the parents).